typescript/packages/dbmate/src/resolveBinary.ts

```diff
@@ -0,0 +1,25 @@
+import { createRequire } from "node:module";
+import { arch, platform } from "node:process";
+
+/**
+ * Resolve path to dbmate for the current platform
+ * */
+export function resolveBinary(): string {
+  const ext = platform === "win32" ? ".exe" : "";
+  const path = `@dbmate/${platform}-${arch}/bin/dbmate${ext}`;
+
+  try {
+    return createRequire(import.meta.url).resolve(path);
```

Comment:
@defunctzombie for future reference, `createRequire()` is the magic needed to access require from esm.

Reply:
😮